File: java/src/org/raml/amf/core/domain/Response.java
package org.raml.amf.core.domain;

import api_modeling_framework.model.domain.ParsedResponse;
import org.raml.amf.core.exceptions.InvalidModelException;
import org.raml.amf.utils.Clojure;

/**
 * Created by antoniogarrote on 04/05/2017.
 */


/**
 * Information about the response returned by an operation, associated to a particular status
 */
public class Response extends GenericOperationUnit {
    static {
        Clojure.require(Clojure.API_MODELING_FRAMEWORK_MODEL_DOMAIN);
    }

    public Response(ParsedResponse rawModel) {
        super(rawModel);
    }

    public Response(String id) {
        super(Clojure.var(
                Clojure.API_MODELING_FRAMEWORK_MODEL_DOMAIN,
                "api-modeling-framework.model.domain/map->ParsedResponse"
                ).invoke(Clojure.map())
        );
        this.setId(id);
    }

    /**
     * Status code for the response
     * @return
     */
    public String getStatusCode() {
        String code = (String) this.wrapped().status_code();
        if (code != null)
            return code;
        else
            return null;
    }

    public void setStatusCode(String status) {
        this.rawModel = Clojure.setKw(this.wrapped(), "status-code", status);
    }

    private ParsedResponse wrapped() {
        return (ParsedResponse) this.rawModel;
    }
}

File: bindings/js/src/core/domain/Response.ts
/**
 * Created by antoniogarrote on 05/05/2017.
 */

import {GenericOperationUnit} from "./GenericOperationUnit";
import {Clojure} from "../../Clojure";
import {DomainModel} from "./DomainModel";

/**
 * Information about the response returned by an operation, associated to a particular status
 */
export class Response extends GenericOperationUnit {

    /**
     * HTTP status code for the response
     * @return
     */
    public getStatusCode(): string | undefined {
        return Clojure.amf_domain.status_code(this.rawModel);
    }

    public setStatusCode(statusCode: string | undefined) {
        this.rawModel = Clojure.amf.update(DomainModel.domain_builder, this.rawModel, Clojure.kw("status-code"), statusCode);
    }

    public static build(id: string): Response {
        return new Response(Clojure.amf.build(
            DomainModel.domain_builder,
            Clojure.amf_domain.map__GT_ParsedResponse,
            id));
    }

}